A110-child-channel-plugins.md

```diff
@@ -0,0 +1,215 @@
+# **A110: Child Channel Configuration Plugins**
+
+* **Author(s)**: [Abhishek Agrawal](mailto:agrawalabhi@google.com)
+* **Status**: Draft
+* **To be implemented in**: Core, Java, Go
+* **Last updated**: 2025-12-23
+
+## **Abstract**
+
+This proposal introduces a mechanism to configure "child channels"—channels created internally by gRPC components (such as xDS control plane channel). Currently, these internal channels cannot easily inherit configuration (like metric sinks and interceptors) from the user application without relying on global state. This design proposes a language-specific approach to injecting configuration: using `functional interfaces` in Java, `DialOptions` in Go, and `ChannelArgs` in Core.
+
+## **Background**
+
+Complex gRPC ecosystems often require the creation of auxiliary channels that are not directly instantiated by the user application. Two primary examples are:
+
+1. **xDS (Extensible Discovery Service)**: When a user creates a channel with an xDS target, the gRPC library internally creates a separate channel to communicate with the xDS control plane. Currently, users have limited ability to configure this internal control plane channel.
+2. **Advanced Load Balancing (RLS, GrpcLB):** Policies like RLS (Route Lookup Service) and GrpcLB, as well as other high-level libraries built on top of gRPC, frequently create internal channels to communicate with look-aside load balancers or backends.
+
+### **The Problem**
+
+There is currently no standardized way to configure behavior for these child channels.
+
+* **Metrics**: Users need to configure metric sinks so that telemetry from internal channels can be read and exported.
+* **Interceptors**: Users may need to apply specific interceptors (e.g., for authentication, logging, or tracing) to internal traffic.
+* **No Global State**: These configurations cannot be set globally (e.g., using static singletons) because different parts of an application may require different configurations, such as different metric backends or security credentials.
```

Comment:
This seems like it should be a separate paragraph rather than a bullet in this list.  It's not an example of what state needs to be propagated to child channels; it's a comment about why global state is not a good way to propagate the state to child channels.